core/spec/models/spree/country_spec.rb

```diff
@@ -55,23 +55,43 @@
     end
   end
 
-  describe 'ensure default country in not deleted' do
-    before { Spree::Config[:default_country_id] = america.id }
-
-    context 'will not destroy country if it is default' do
-      subject { america.destroy }
+  describe 'ensure proper country deletion' do
+    context 'when deleting default country' do
+      before { Spree::Config[:default_country_id] = america.id }
 
-      it { is_expected.to be_falsy }
+      it 'does not destroy country' do
+        expect(america.destroy).to be_falsy
+      end
 
-      context 'error should be default country cannot be deleted' do
-        before { subject }
+      it 'sets correct error message' do
+        america.destroy
 
-        it { expect(america.errors[:base]).to include(Spree.t(:default_country_cannot_be_deleted)) }
+        expect(america.errors[:base]).to include(Spree.t(:default_country_cannot_be_deleted))
       end
     end
 
-    context 'will destroy if it is not a default' do
-      it { expect(canada.destroy).to be_truthy }
+    context 'when deleting not a default country' do
+      context 'when country has no dependent addresses' do
+        it 'destroys country successfully' do
+          expect(canada.destroy).to be_truthy
+        end
+      end
+
+      context 'when country has dependent addresses' do
+        before do
+          create(:ship_address, country: canada, zipcode: 'l3l 4p4')
+        end
+
+        it 'does not destroy country' do
+          expect(canada.destroy).to be_falsy
+        end
+
+        it 'sets correct error message' do
+          canada.destroy
+
+          expect(canada.errors[:base]).to include('Cannot delete record because dependent addresses exist')
```

Comment:
It's not quite right way to test error messages, though I was not able to figure out which ActiveRecord's translation symbol it is. Any ideas how to improve it?